Phase_01/Phase_01_Full_Text_Search/src/Main.java

```diff
@@ -1,139 +1,53 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
 
 public class Main {
-    private static ArrayList<Set<String>> documents;
-    private static int inputCount;
-    private static boolean containsStopWords;
-    private static ArrayList<String> normalWords;
-    private static ArrayList<String> plusWords;
-    private static ArrayList<String> minusWords;
-    private static InvertedIndex ii;
-    private static ArrayList<Integer> searchResult;
-
+    /**
+     * To use the full text search library, first you should construct
+     * an instance of FullTextSearch class and give the path to your documents'
+     * folder and normalization method and tokenizer. then you can search by
+     * calling the Search method on your FullTextSearch instance.
+     */
     public static void main(String[] args) throws IOException {
-        // construct inverted index for documents
-        documents = FileReader.ReadBooks("..\\Books");
-        List<String> documentsNames = FileReader.lastReadNames;
-        ii = new InvertedIndex(documents);
-        // input and search
+        // construct the FullTextSearch instance.
+        FullTextSearch fts = new FullTextSearch(new RemoveMarksAndUpperCaseNormalization(),
+                new StringTokenizer(" "));
+
+        // read the documents in the folder and add them to search data.
+        for (Document d : FileReader.GetDocumentsInFolder("..\\Books", "txt")){
+            fts.AddDocument(d);
+        }
+
         System.out.println("if you want to stop the program enter empty");
         while (true) {
-            inputCount = 0;
-            containsStopWords = false;
-            if (!ReadInput()){
+            // get search input from the user
+            System.out.println("Input your search phrase:");
+            BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(System.in));
+            String input = reader.readLine();
+
```

Comment:
suggest to add the option of giving the query inside of the code in addition of console